cmd/bucket-handlers.go

```diff
@@ -692,6 +693,16 @@ func (api objectAPIHandlers) DeleteBucketHandler(w http.ResponseWriter, r *http.
 		return
 	}
 
+	// Notify all peers (including self) to update in-memory state
+	for addr, err := range globalNotificationSys.UpdateBucketPolicy(bucket) {
+		errorIf(err, "unable to update policy change in remote peer %v", addr)
+	}
+
+	globalNotificationSys.RemoveNotification(bucket)
+	for addr, err := range globalNotificationSys.DeleteBucket(bucket) {
```

Comment:
`globalNotificationSys.DeleteBucket()` is not only removing notification configuration, any actions can be taken this `DeleteBucket` event.  Currently `PeerRPCReceiver` removes notification, but it will be leveraged to do more actions like calling policy subsystem to remove bucket from policy map.

Reply:
currently who removes the bucket from policy map?

Reply:
Not now. I will be done that way in my refactor of policy handling.